File: src/main/java/com/vaadin/demo/component/tooltip/TooltipBasic.java
package com.vaadin.demo.component.tooltip;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("tooltip-basic")
public class TooltipBasic extends Div {

  public TooltipBasic() {
    // tag::snippet[]
    TextField textField = new TextField();
    textField.setPlaceholder("Search");
    textField.setPrefixComponent(new Icon("lumo", "search"));
    // textField.setTooltip("Wrap in “quotes” for exact phrase");
    add(textField);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<TooltipBasic> { // hidden-source-line
  } // hidden-source-line
}

File: frontend/demo/component/tooltip/tooltip-basic.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/icon';
import '@vaadin/text-field';
import '@vaadin/tooltip';
import '@vaadin/vaadin-lumo-styles/vaadin-iconset.js';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('tooltip-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-text-field placeholder="Search">
        <vaadin-icon slot="prefix" icon="lumo:search"></vaadin-icon>
        <vaadin-tooltip slot="tooltip" text="Wrap in “quotes” for exact phrase"></vaadin-tooltip>
      </vaadin-text-field>
      <!-- end::snippet[] -->
    `;
  }
}
